javascript/extractor/BUILD.bazel

```diff
@@ -0,0 +1,18 @@
+load("@//:common.bzl", "codeql_java_project")
+
+codeql_java_project(
+    name = "extractor",
+    deps = [
+        "@//extractor",
+        "@//resources/lib/java:commons-compress",
+        "@//resources/lib/java:gson",
+        "@//resources/lib/java:jericho-html",
+        "@//resources/lib/java:slf4j-api",
+        "@//resources/lib/java:snakeyaml",
+        "@//resources/lib/java/DO_NOT_DISTRIBUTE:junit",
```

Comment:
Do we actually need this in `deps`?

Reply:
Surprisingly, yes. I think the project could use a better setup, to not compile (and ship) the unit tests (see https://github.com/github/codeql/tree/main/javascript/extractor/src/com/semmle/js/extractor/test).
I don't know how or where those tests are run either.

Reply:
> I don't know how or where those tests are run either.

That seems important to find out, if only to ensure that we don't break anything by accident.

Reply:
I found out how it's working - there's a separate build target that runs these tests, and it's not broken by either this PR or the internal PR.